Add support for parameter `rename` on ipahostgroup.

rjeffman opened the pull request:
FreeIPA 4.8.7 introduced an option to rename an existing hostgroup.
This patch adds support for renaming hostgroups if the option is
available on installed IPA version.

rjeffman commented:
This patch has been tested against Fedora 32 with with `updates-testing` and the following RPM versions:
* freeipa-selinux-4.8.7-1.fc32.noarch
* freeipa-client-common-4.8.7-1.fc32.noarch
* freeipa-server-4.8.7-1.fc32.x86_64
* freeipa-healthcheck-core-0.5-3.fc32.noarch
* freeipa-common-4.8.7-1.fc32.noarch
* freeipa-client-4.8.7-1.fc32.x86_64
* freeipa-server-common-4.8.7-1.fc32.noarch
* freeipa-server-dns-4.8.7-1.fc32.noarch

rjeffman commented:
I'm marking this PR as `blocked`, until the required version is available on a `stable` Fedora repo.

t-woerner commented:
I would more like to have a new state `renamed`. State `present` with `rename` feels strange.

t-woerner commented:
A rename and also further changes at the same time is unexpected. The rename should be something special as it can not be idempotent.

If a rename and want to have other changes should be done at the same time... what will be the result
1) If there is a group to rename
2) If the group has been renamed already
3) If there is no such group
4) If the target group already exists altogether with the source group

rjeffman commented:
Added state `renamed` to the module to handle hostgroup renaming.

t-woerner commented on tests/hostgroup/test_hostgroup_rename.yml:
```diff
@@ -0,0 +1,99 @@
+    register: result
+    failed_when: result.changed
+
+  - name : Rename host-group from `databases` to `datalake`
```
There is a space before ":".

rjeffman replied:
fixed.

t-woerner commented on tests/hostgroup/test_hostgroup_rename.yml:
```diff
@@ -0,0 +1,99 @@
+    register: result
+    failed_when: not result.changed
+
+  - name : Rename host-group from `databases` to `datalake`, again
```
There is a space before ":".

rjeffman replied:
fixed

t-woerner commented on tests/hostgroup/test_hostgroup_rename.yml:
```diff
@@ -0,0 +1,99 @@
+    register: result
+    failed_when: result.changed
+
+  - name : Rename host-group with same name.
```
There is a space before ":".

rjeffman replied:
fixed

t-woerner commented on tests/hostgroup/test_hostgroup_rename.yml:
```diff
@@ -0,0 +1,99 @@
+    register: result
+    failed_when: result.changed
+
+  - name : Rename inexistent host-group.
```
There is a space before ":".

rjeffman replied:
fixed

t-woerner commented:
Should a repeated use of `action: renamed` make sure that a hostgroup with the new name exists additionally?

rjeffman commented:
The current implemented behavior is:
* Rename if `name` exists and `rename` doesn't. (result.changed is True)
* Do nothing if `name` does not exist and `rename` does, or if `name` equals to `rename`. (result.changed is False)
* Fail if neither or both `name` and `rename` exist (result.failed is True).

t-woerner commented:
The test failure is due to not supporting the rename in IPA versions prior to 4.7.8. Therefore this error needs to be ignored here.
